File: src/app/src/main/java/com/huawei/podcast/java/data/model/PodCastList.java
/**
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.huawei.podcast.java.data.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.huawei.podcast.java.utils.Constants;

import java.io.Serializable;

@Entity(tableName = Constants.PLAYBACK_HISTORY)
public class PodCastList implements Serializable {
    @PrimaryKey
    @ColumnInfo
    int id;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    @ColumnInfo
    String description;

    @ColumnInfo(name = Constants.URL)
    String url;

    @ColumnInfo(name = Constants.AUTHOR)
    String author;

    @ColumnInfo(name = Constants.DATE)
    String date;

    @ColumnInfo(name = Constants.TITLE)
    String title;

    @ColumnInfo(name = Constants.CATEGORY)
    String category;

    @ColumnInfo(name = Constants.POSITION)
    int position;
}

File: src/app/src/main/java/com/huawei/podcast/kotlin/data/model/PodCastList.kt
/**
 * Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.huawei.podcast.kotlin.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.huawei.podcast.kotlin.utils.Constants
import java.io.Serializable

@Entity(tableName = Constants.TABLE_NAME)
data class PodCastList(
    @PrimaryKey
    @ColumnInfo val id: Int?,
    @ColumnInfo val description: String?,
    @ColumnInfo(name = Constants.URL) val url: String?,
    @ColumnInfo(name = Constants.AUTHOR) val author: String?,
    @ColumnInfo(name = Constants.DATE) var date: String?,
    @ColumnInfo(name = Constants.TITLE) var title: String?,
    @ColumnInfo(name = Constants.CATEGORY) val category: String?,
    @ColumnInfo(name = Constants.POSITION) val position: Int
) : Serializable
